nodeup/pkg/model/kubeapiserver.go

```diff
@@ -127,22 +141,30 @@ func (b *KubeAPIServerBuilder) writeAuthenticationConfig(c *fi.ModelBuilderConte
 			Type:     nodetasks.FileType_File,
 		}
 		c.AddTask(t)
+
 		return nil
-	} else {
-		return fmt.Errorf("Unrecognized authentication config %v", b.Cluster.Spec.Authentication)
 	}
+
+	return fmt.Errorf("Unrecognized authentication config %v", b.Cluster.Spec.Authentication)
 }
 
+// buildPod is responsible for generating the kube-apiserver pod and thus manifest file
 func (b *KubeAPIServerBuilder) buildPod() (*v1.Pod, error) {
 	kubeAPIServer := b.Cluster.Spec.KubeAPIServer
-
 	kubeAPIServer.ClientCAFile = filepath.Join(b.PathSrvKubernetes(), "ca.crt")
 	kubeAPIServer.TLSCertFile = filepath.Join(b.PathSrvKubernetes(), "server.cert")
 	kubeAPIServer.TLSPrivateKeyFile = filepath.Join(b.PathSrvKubernetes(), "server.key")
-
 	kubeAPIServer.BasicAuthFile = filepath.Join(b.PathSrvKubernetes(), "basic_auth.csv")
 	kubeAPIServer.TokenAuthFile = filepath.Join(b.PathSrvKubernetes(), "known_tokens.csv")
 
+	// @check if we are using secure kubelet client certificates
+	if b.UseSecureKubelet() {
+		// @note we are making assumption we are using the one's created by the pki model, not custom defined ones
```

Comment:
Good to call this out.  This is tricky actually.  In some places we haven't mapped the flag in the model, and we just directly add the flag to the arguments.  But that feels sort of hacky in that then we aren't using the flag code we've worked so hard on!

Realistically, eventually with componentconfig in k8s these will move out of flags (either to secrets or to dynamically generated and locally stored certificates, using the CertificateSigningRequest API).  But we aren't there yet.

One option could be to `glog.Warningf` if the values are not empty.  But this is fine I think... setting the kubelet / kubeapiserver config flags directly is "expert mode" kops and it is then possible/easy to break things (for typical usage we prefer setting options which set these flags, which is how kops continues to work across k8s versions even if the underlying flags or flag semantics change).